examples/c/src/ivf_pq_c_example.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuvs/core/c_api.h>
+#include <cuvs/neighbors/ivf_pq.h>
+#include <cuvs/neighbors/refine.h>
+
+#include <cuda_runtime.h>
+#include "common.h"
+
+void ivf_pq_build_search(cuvsResources_t *res, DLManagedTensor * dataset_tensor, DLManagedTensor * queries_tensor) {
+    // Create default index params
+    cuvsIvfPqIndexParams_t index_params;
+    cuvsIvfPqIndexParamsCreate(&index_params);
+    index_params->n_lists                  = 1024; // default value
+    index_params->kmeans_trainset_fraction = 0.1;
+    //index_params->metric default is L2Expanded
+    index_params->pq_bits = 8;
+    index_params->pq_dim = 2;
+
+    // Create IVF-PQ index
+    cuvsIvfPqIndex_t index;
+    cuvsIvfPqIndexCreate(&index);
+
+    printf("Building IVF-PQ index\n");
+
+    // Build the IVF-PQ Index
+    cuvsError_t build_status = cuvsIvfPqBuild(*res, index_params, dataset_tensor, index);
+    if (build_status != CUVS_SUCCESS) {
+        printf("%s.\n", cuvsGetLastErrorText());
+        cuvsIvfPqIndexDestroy(index);
+        cuvsIvfPqIndexParamsDestroy(index_params);
+        return;
+    }
+    
+    // Create output arrays.
+    int64_t topk      = 10;
+    int64_t n_queries = queries_tensor->dl_tensor.shape[0];
+
+    //Allocate memory for `neighbors` and `distances` output
+    int64_t *neighbors_d;
+    float *distances_d;
+    cuvsRMMAlloc(*res, (void**) &neighbors_d, sizeof(int64_t) * n_queries * topk);
+    cuvsRMMAlloc(*res, (void**) &distances_d, sizeof(float) * n_queries * topk);
+
+    DLManagedTensor neighbors_tensor;
+    int64_t neighbors_shape[2] = {n_queries, topk};
+    int_tensor_initialize(neighbors_d, neighbors_shape, &neighbors_tensor);
+
+    DLManagedTensor distances_tensor;
+    int64_t distances_shape[2] = {n_queries, topk};
+    float_tensor_initialize(distances_d, distances_shape, &distances_tensor);
+
+    // Create default search params
+    cuvsIvfPqSearchParams_t search_params;
+    cuvsIvfPqSearchParamsCreate(&search_params);
+    search_params->n_probes = 50;
+    search_params->internal_distance_dtype = CUDA_R_16F;
+    search_params->lut_dtype = CUDA_R_16F;
+
+    // Search the `index` built using `cuvsIvfPqBuild`
+    cuvsError_t search_status = cuvsIvfPqSearch(*res, search_params, index,
+     queries_tensor, &neighbors_tensor, &distances_tensor);
+    if (build_status != CUVS_SUCCESS) {
```

Comment:
You probably meant to check the search_status here ? (build_status was checked above and is guaranteed to be success)

```suggestion
    if (search_status != CUVS_SUCCESS) {
```

Reply:
Also - should we just abort the example if the search failed ?

Reply:
Yes. So I modified and add  'exit(-1);'